packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -378,216 +387,122 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
       selection: TextSelection.collapsed(offset: selectionString.length),
       text: selectionString,
     );
-    _updateActions();
-    _updateOverlay();
-    widget.onSelected?.call(_selection!);
+    widget.onSelected?.call(nextSelection);
+    _updateOptionsViewVisibility();
   }
 
   void _updateHighlight(int newIndex) {
     _highlightedOptionIndex.value = _options.isEmpty ? 0 : newIndex % _options.length;
   }
 
   void _highlightPreviousOption(AutocompletePreviousOptionIntent intent) {
-    if (_userHidOptions) {
-      _userHidOptions = false;
-      _updateActions();
-      _updateOverlay();
-      return;
-    }
+    assert(_canShowOptionsView);
+    _updateOptionsViewVisibility();
+    assert(_optionsViewController.isShowing);
     _updateHighlight(_highlightedOptionIndex.value - 1);
   }
 
   void _highlightNextOption(AutocompleteNextOptionIntent intent) {
-    if (_userHidOptions) {
-      _userHidOptions = false;
-      _updateActions();
-      _updateOverlay();
-      return;
-    }
+    assert(_canShowOptionsView);
+    _updateOptionsViewVisibility();
+    assert(_optionsViewController.isShowing);
     _updateHighlight(_highlightedOptionIndex.value + 1);
   }
 
   Object? _hideOptions(DismissIntent intent) {
-    if (!_userHidOptions) {
-      _userHidOptions = true;
-      _updateActions();
-      _updateOverlay();
+    if (_optionsViewController.isShowing) {
+      _optionsViewController.hide();
       return null;
-    }
-    return Actions.invoke(context, intent);
-  }
-
-  void _setActionsEnabled(bool enabled) {
-    // The enabled state determines whether the action will consume the
-    // key shortcut or let it continue on to the underlying text field.
-    // They should only be enabled when the options are showing so shortcuts
-    // can be used to navigate them.
-    _previousOptionAction.enabled = enabled;
-    _nextOptionAction.enabled = enabled;
-    _hideOptionsAction.enabled = enabled;
-  }
-
-  void _updateActions() {
-    _setActionsEnabled(_focusNode.hasFocus && _selection == null && _options.isNotEmpty);
-  }
-
-  bool _floatingOptionsUpdateScheduled = false;
-  // Hide or show the options overlay, if needed.
-  void _updateOverlay() {
-    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
```

Comment:
And great that we no longer have to deal with SchedulerBinding stuff :star_struck: